### PR TITLE
feat: [CI-5559]: Add testSplitStrategy and enableTestSplitting to RunTests step

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37600)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/beans/serializer/RunTimeInputHandler.java [hunk 1]
```diff
@@ -113,6 +114,14 @@ public static String resolveBuildTool(ParameterField<TIBuildTool> buildTool) {
     }
   }
 
+  public static String resolveSplitStrategy(ParameterField<TISplitStrategy> splitStrategy) {
+    if (splitStrategy == null || splitStrategy.isExpression() || splitStrategy.getValue() == null) {
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/main/java/io/harness/ci/beans/serializer/RunTimeInputHandler.java:118]: Does this mean split strategy cannot be an expression?

Comment (rutvijmehta-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/beans/serializer/RunTimeInputHandler.java:118]: Yes, we only allow values and run time inputs for our drop-downs (example: Build tool)

**Comment by rutvijmehta-harness** [comment]
trigger messagecheck

**Review by jamie-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37600)* status: 1. unresolved discussions.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/RunTestsStepProtobufSerializer.java [hunk 2]
```diff
@@ -140,6 +140,11 @@ public UnitStep serializeStepWithStepParameters(RunTestsStepInfo runTestsStepInf
     }
     long timeout = TimeoutUtils.getTimeoutInSeconds(parameterFieldTimeout, runTestsStepInfo.getDefaultTimeout());
 
+    runTestsStepBuilder.setParallelizeTests(resolveBooleanParameter(runTestsStepInfo.getEnableTestSplitting(), false));
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/RunTestsStepProtobufSerializer.java:143]: Let's use either `split` or `parallelize` across the board. It seems split is popular in the codebase and yaml.

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37600)* status: :shipit: complete! ✓

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/RunTestsStepProtobufSerializer.java (hunk 2 above)
Comment (rutvijmehta-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/RunTestsStepProtobufSerializer.java:143]: The addon image is already built. We can update the proto and CI Manager in a follow up change.

**Pull request merged**